src/http.rs

```diff
@@ -80,13 +109,19 @@ impl Client {
     }
 
     pub async fn fetch_stream(&self, url: &str) -> Result<Download> {
-        debug!("Downloading {:?}...", url);
-        let resp = self.send_get(url).await?;
-
-        let total = resp.content_length().unwrap_or(0);
+        let total = self
+            .send_head(url)
+            .await?
+            .headers()
+            .get(CONTENT_LENGTH)
+            .map(|val| val.to_str())
+            .context("missing CONTENT_LENGTH")?
+            .map(str::parse)??;
+        debug!("{total} bytes to download");
```

Comment:
Yeah, that's fair. We probably don't need both functions, would you be okay with removing `fn send_get` (and `fn fetch`)?

Reply:
Sorry for the late reply, I had some vague memories why this was needed but didn't have time to look into it and write it down (thanks for the ping on irc 😺):

Essentially `send_get` is opening the connection and sending the request, but never processing the response. This is because there are two functions making use of `send_get` and then handle the response differently:

- `fetch`: Sends the request and reads the entire response into a `Vec<u8>` with a timeout for the total duration of the request.
- `fetch_stream`: Sends the request, checks the `Content-Length` header and then sets up the `Download` object that can then read the response in chunks, the timeout is reset for every chunk and is effectively "timeout on inactivity", if no data was received from the connection anymore for a certain amount of time.

Implementing apt involves at least 3 requests, the first two are quite small and are a good fit for `fetch`, the last one (the download of the actual .deb) is a better fit for `fetch_stream` (especially since we want to implement download resumption).